include/deal.II/numerics/vector_tools.templates.h

```diff
@@ -1205,6 +1494,9 @@ namespace VectorTools
                 const hp::QCollection<dim-1>               &q_boundary,
                 const bool                                  project_to_boundary_first)
   {
+    Assert((dynamic_cast<const parallel::Triangulation<dim,spacedim>* > (&(dof.get_triangulation()))==0),
```

Comment:
why is it not implemented? Even if you do `MatrixFree`, you don't have to have a parallel Triangulation. You just skip setting `mpi_communicator`. I think this Assert should go.